src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -5232,20 +5231,19 @@ internal static Expression GetTargetExpr(DynamicMetaObject target, Type castToTy
             }
 
             var type = castToType ?? ((value != null) ? value.GetType() : typeof(object));
-#if CORECLR
-            var typeInfo = type.GetTypeInfo();
+
             // Assemblies in CoreCLR might not allow reflection execution on their internal types. In such case, we walk up
             // the derivation chain to find the first public parent, and use reflection methods on the public parent.
-            if (!TypeResolver.IsPublic(typeInfo) && DotNetAdapter.DisallowPrivateReflection(typeInfo))
+            if (!TypeResolver.IsPublic(type) && DotNetAdapter.DisallowPrivateReflection(type))
```

Comment:
The same.

Reply:
Can you please elaborate? I don't understand why this would be a breaking change.